### PR TITLE
Improve canvas/board image logic

### PR DESCRIPTION
**Pull request by JAKONL**
Spent some time reworking how the code to obtain the current board works. Now the `get_unset_pixel` function should no longer think the entire right side of the canvas is black!

### CODE REVIEW

**Comment by Plantchant** [comment]
This has already been fixed

**Comment by JAKONL** [comment]
@Plantchant - that's why I commented out the bit that saves it to disk :)

This new get_board code improves the logging of the get_board system and the WebSocket communication to be more flexible when the board changes. 

There are fewer hardcoded values.

**Comment by Plantchant** [comment]
> @Plantchant - that's why I commented out the bit that saves it to disk :)
> 
> This new get_board code improves the logging of the get_board system and the WebSocket communication to be more flexible when the board changes.
> 
> There are fewer hardcoded values.

Much better with that commented out. Your code looks prettier so as long as it works, good job.

**Review by RomainL972 — CHANGES_REQUESTED** [review]
Can you remove dead code please ?

**Comment by JAKONL** [comment]
Hey @RomainL972 - I'm having trouble finding the dead code you're referencing. Would you mind pointing me in the right direction?

File: main.py [hunk 1]
```diff
@@ -251,37 +259,24 @@ def get_board(self, access_token_in):
...
 
-        image_sizex = 2
-        image_sizey = 1
+        while True:
+            canvas_payload = json.loads(ws.recv())
+            if canvas_payload["type"] == "data":
+                canvas_details = canvas_payload["payload"]["data"]["subscribe"]["data"]
+                logger.debug("Canvas config: {}", canvas_payload)
+                break
+        #logger.debug("Canvas Details: {}", canvas_details)
```
Comment (RomainL972) [inline main.py:269]: Dead code

File: main.py [hunk 2]
```diff
@@ -304,39 +299,56 @@ def get_board(self, access_token_in):
...
-        new_img = Image.new("RGB", (1000 * 2, 1000))
-
-        x_offset = 0
-        for img in imgs:
-            new_img.paste(img, (x_offset, 0))
-            x_offset += img.size[0]
-
-        logger.info("Got image.")
-        # new_img.show()
+        #TODO: Multiply by canvas_details["canvasConfigurations"][i]["dx"] and canvas_details["canvasConfigurations"][i]["dy"] instead of hardcoding it
```
Comment (RomainL972) [inline main.py:335]: Dead code

File: main.py [hunk 3]
```diff
@@ -304,39 +299,56 @@ def get_board(self, access_token_in):
...
+        dx_offset = 0
+        for idx, img in enumerate(imgs):
+            logger.debug("Adding image: {}", img)
+            dx_offset = int(canvas_details["canvasConfigurations"][idx]["dx"])
+            new_img.paste(img, (dx_offset, 0))
+            
+        logger.debug("Saving image to disk as board.png")
+        
+        #Uncomment the line below to save the canvas image to disk
+        #new_img.save("board.png")
```
Comment (RomainL972) [inline main.py:351]: Dead code

**Review by RomainL972 — COMMENTED** [review]
Commented code is usually dead code and should be avoided if possible

File: main.py (hunk 2 above)
Comment (JAKONL) [reply main.py:335]: This is a reminder to improve the multiplication logic below when/if the canvas grows.

Comment (RomainL972) [reply main.py:335]: Ok for the todo

File: main.py (hunk 3 above)
Comment (JAKONL) [reply main.py]: Removed

File: main.py (hunk 1 above)
Comment (JAKONL) [reply main.py]: Removed in [cfe84b4](https://github.com/rdeepak2002/reddit-place-script-2022/pull/209/commits/cfe84b49f97615afb161928676876551507fecbf)

File: main.py [hunk 4]
```diff
@@ -304,39 +298,54 @@ def get_board(self, access_token_in):
...
+        logger.debug("New image height: {}", new_img_height) 
+        
+        new_img = Image.new("RGB", (new_img_width, new_img_height))
+        dx_offset = 0
+        for idx, img in enumerate(imgs):
+            logger.debug("Adding image: {}", img)
+            dx_offset = int(canvas_details["canvasConfigurations"][idx]["dx"])
+            new_img.paste(img, (dx_offset, 0))
+            
+        logger.debug("Saving image to disk as board.png")
```
Comment (RomainL972) [inline main.py:347]: Can we remove that line ?

**Review by RomainL972 — APPROVED** [review]
Should be all good now ! If you're sure it works I can merge this

EDIT : First the latest changes must be merged. Also since it's a big change I'll ask for at least one more review.

File: main.py (hunk 4 above)
Comment (JAKONL) [reply main.py]: Removed in [603e5fb](https://github.com/rdeepak2002/reddit-place-script-2022/pull/209/commits/603e5fb5a8910338c03cf831e90b4dafd10f05f8)

**Review by robiot — APPROVED** [review]
Lgtm. Tho merge conflicts again aaaah.

**Comment by robiot** [comment]
Ok perfect

**Comment by JAKONL** [comment]
Thanks @RomainL972 @robiot for the assist here

**Pull request merged**